### PR TITLE
feat:  mailchimp email service

### LINKED ISSUE #1153: Storage limit - Add MailChimp email service
### Overarching user needs / context:
* Users need to receive warning emails from support@web3.storage about their storage quota with specific messages and some dynamically generated data on how much they are using and what their current quota is.
* Admins (support@web3.storage) need to receive an email when a user has gone over their allowed storage quota.

### Scope of this ticket:
This is a subtask of #990. We want to take the `EmailService` class that's been stubbed out in that ticket and make it actually connect to MailChimp and send the emails.

* We'll use the _templated_ transactional emails, so that we can use the visual template editor in MailChimp and therefore don't have to deal with the nightmare of trying to create HTML emails that will work across different email clients; instead our API calls are just `"Send 'template_x' to 'user@example.com' with variables { ... }"`.
* We'll make this as a mini sub-package inside the "cron" package, but it will be entirely generic, so could be shifted out to somewhere else to make it reusable by the other packages.
* Calls to send emails will go through the `EmailSerivce` class, and MailChimp will be plugged in behind this, so in future we could switch to a different email sending provider without changing any of the rest of the application code.

### Acceptance criteria:
- [ ] Once we run this service (on-demand), it needs to take all users returned by the checks and send them the appropriate email with the correct details on their storage usage and how to request an increase.

### PR DESCRIPTION
**Pull request by adamalton**
Fixes #1153.

This is pretty much there, but needs some tests adding for it, and also needs some stubs adding (or some way of avoiding trying to actually send emails) for the existing tests to call it done.

For avoiding actually trying to send emails for the existing tests, there are a couple of possible routes here, neither of which I've had time to implement.

There's an idea of a dummy email provider class, which we could use. It would be a class which implements the `sendEmail` method as a no-op method that doesn't do anything. The hook for using this is already implemented (as you'll see at the bottom of the `service.js` file. It just needs the actual class adding. It can then be used by either setting `EMAIL_PROVIDER=dummy` in the environment, or by passing `{ provider: 'dummy' }` into the `EmailService` constructor.

Alternatively, we could use `sinon` to stub the `sendEmail` method of the `MailchimpEmailProvider` class.

I don't have a strong opinion on which to go for.

Another thing which I'm not quite happy with is the scattered array of definitions of email types! I feel like it needs a bit of a tidy up and consolidation, but I haven't got as far as doing that.

Here are screenshots of how the emails look at the moment. If the copy needs tweaking then instructions for how to edit the templates are in the README file that I've added:

**User alerts for 75%/80%/85%/90%:**
75-quota-reached

**User alert for over 100%:**
storage-quota-exceeded

**Admin list:**
admin-list-of-users

### CODE REVIEW

File: packages/cron/src/lib/email/types.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+   */
+  formatTemplateVars (vars) {
+    return vars
+  }
+}
+
+export class User75PercentStorage extends EmailType {
+  constructor () {
+    super()
+    this.subject = '75% of storage quota reached'
```
[the remark this answers is not in the document]
Comment (adamalton) [reply packages/cron/src/lib/email/types.js:41]: Given that our `EMAIL_FROM_NAME` is `"Web3.Storage"`, I'm thinking that that will make it sufficiently clear in someone's inbox, no?

File: packages/cron/src/lib/email/service.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    // Send the email
+    log(`📧 Sending email '${emailType}' to ${user.name} (${user.email}).`)
+    const messageId = await this.provider.sendEmail(emailType, user.email, user.name, formattedVars)
+    if (messageId) {
+      // Log the email
+      await this.db.logEmailSent({ userId: user._id, emailType, messageId })
+    } else {
+      const msg = `📧 🚨 Failed to send ${emailType} email to ${user.name} (${user.email}) using ${this.providerStr}.`
+      if (failSilently) {
+        log(msg)
```
[the remark this answers is not in the document]
Comment (adamalton) [reply packages/cron/src/lib/email/service.js:79]: Good point, we should still log the error. I'll change it.

[the remark this answers is not in the document]
Comment (adamalton) [reply packages/cron/src/lib/email/service.js]: Done

File: packages/cron/src/lib/email/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+Email
```
[the remark this answers is not in the document]
Comment (adamalton) [reply packages/cron/src/lib/email/README.md:1]: Woo!

**Comment by adamalton** [comment]
I'm calling this done. Just need to solve the mystery of the failing `api` tests.

File: packages/cron/src/lib/email/service.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    this.providerStr = provider || process.env.EMAIL_PROVIDER || 'mailchimp'
+    this.provider = new EMAIL_PROVIDERS[this.providerStr]()
+  }
+
+  /**
+   * Send an email to a user.
+   * Optionally checks email sending history for this user and email type to avoid
+   * re-sending if user has been recently notified.
+   * @param {{_id: number, email: string, name: string}} user
+   * @param {string} emailType
```
[the remark this answers is not in the document]
Comment (adamalton) [reply packages/cron/src/lib/email/service.js:43]: I've left the types as they are for the moment. As you mentioned IRL, maybe we want to tidy them up as a follow-up task, but probably not worth doing right now.

File: packages/cron/src/lib/email/providers/mailchimp.js [hunk 5]
```diff
@@ -24,7 +25,8 @@ export default class MailchimpEmailProvider {
 
   /**
      * Sends the email of the given type to the given recipient.
-     * Returns the unique message ID if successful, or undefined if the sending failed.
+     * Returns the unique message ID.
+     * Throws EmailSendError if the sending failed.
```
Comment (flea89) [inline packages/cron/src/lib/email/providers/mailchimp.js:29]: Is this function now only returning Promise<string|undefined>? (line 36)

Comment (adamalton) [reply packages/cron/src/lib/email/providers/mailchimp.js:29]: Done

**Pull request merged**